pkg/oauthserver/oauth/external/github/github.go

```diff
@@ -82,17 +112,35 @@ func NewProvider(providerName, clientID, clientSecret string, organizations, tea
 		}
 	}
 
-	return &provider{
+	p := &provider{
 		providerName:         providerName,
 		clientID:             clientID,
 		clientSecret:         clientSecret,
 		allowedOrganizations: allowedOrganizations,
 		allowedTeams:         allowedTeams,
+		transport:            transport,
+		githubAuthorizeURL:   defaultGithubAuthorizeURL,
+		githubTokenURL:       defaultGithubTokenURL,
+		githubUserApiURL:     defaultGithubUserApiURL,
+		githubUserOrgURL:     defaultGithubUserOrgURL,
+		githubUserTeamURL:    defaultGithubUserTeamURL,
+		githubUserEmailURL:   defaultGithubUserEmailURL,
+	}
+
+	if len(hostname) != 0 {
```

Comment:
maybe switch to

```
if len(hostname) > 0 {
... set custom endpoints ...
}else {
... set standard endpoints ...
}
```

that lessens the chance we end up with a standard endpoint left in the struct that should have been a custom one

Reply:
But that does not actually help to make sure you set it correctly.  The changes required for a new API are now:

1. create two new constants
1. create new field
1. set field in two different places

The unit test was a (probably futile) attempt to make the last step less error prone.

Reply:
it does save you from setting the field in one place, not realizing you needed to override it in the other, and accidentally sending a token to github